docs/docs-beta/docs/guides/declarative-automation.md

```diff
@@ -1,4 +1,136 @@
 ---
 title: "Declarative automation"
+description: "Learn to automatically materialize assets when criteria are met."
+sidebar_position: 10
 sidebar_label: "Declarative automation"
----
\ No newline at end of file
+---
+
+# Declarative Automation
+
+TODO: ADD EXPERIMENTAL CALLOUT
+
+Dagster can automatically materialize assets when criteria are met, enabling a declarative approach to asset materialization. Instead of defining explicit workflows to materialize assets, you describe the conditions under which they should be materialized and let the system kick off runs in response.
+
+For example, you have an asset that's scheduled to execute every day at midnight. Instead of executing whether there's new data or not, you can use Declarative Automation to materialize the asset only after its parent has been updated.
+
+Using Declarative Automation helps you:
+
+- Ensure you're working with the most up-to-date data
+- Optimize resource usage by only materializing assets when needed
+- Simplify how your team understands their assets by consolidating all asset logic to a single location
+
+<details>
+  <summary>Prerequisites</summary>
+
+To follow the steps in this guide, you'll need familiarity with:
+
+- [Assets](/concepts/assets)
+- [Sensors](/todo)
+- [Code locations](/todo)
+
+</details>
+
+## Define automation conditions
+
+Automation conditions describe the conditions under which an asset should be executed. They can be set on native Dagster assets (`@asset`) or [external assets](/guides/external-assets) (`AssetSpec`).
+
+```python
+from dagster import dg
+
+# Condition that materializes an asset that's never been materialized, or
```

Comment:
I think we should move the comment about this condition outside the code block. It's a little confusing reading it